src/scripts.js

```diff
@@ -36,87 +21,89 @@ let pantryArea = document.querySelector('.pantry-cards');
 let recipeSearch = document.querySelector('.search-recipes-input')
 // let cookbook = new Cookbook(recipeData);
 recipeSearch.addEventListener('keyup', ()=>{
-  domUpdates.displayAllRecipes(filterInputs(recipeSearch.value,ingredientsData),favorites)
+  if(favButton.classList.contains('clicked')){
+    domUpdates.displayAllRecipes(filterInputs(recipeSearch.value,ingredientsData,'favoriteRecipes'),currentUser)
+
+  }
+  else if(menuMyUpcomingRecipes.classList.contains('clicked')){
+    domUpdates.displayAllRecipes(filterInputs(recipeSearch.value,ingredientsData,'recipesToCook'),currentUser)
+
+  }
+  else{
+  domUpdates.displayAllRecipes(filterInputs(recipeSearch.value,ingredientsData),currentUser)
+  }
 })
 favButton.addEventListener('click', ()=>{
-  if(!favButton.classList.contains('clicked')){
-  domUpdates.displayFavorites(findFavorites(favorites,recipeDatas))
-  favButton.classList.add('clicked')
+  if(!favButton.classList.contains('clicked')) {
+    domUpdates.displayFavorites(findFavorites(currentUser,recipeDatas,'favoriteRecipes'))
+    favButton.classList.add('clicked')
   }
   else{
     favButton.classList.remove('clicked')
-    domUpdates.displayAllRecipes(recipeDatas,favorites)
+    domUpdates.displayAllRecipes(recipeDatas,currentUser)
   }
 })
-let user, pantry;
+menuMyUpcomingRecipes.addEventListener('click',() =>{
+      if(!menuMyUpcomingRecipes.classList.contains('clicked')) {
+        domUpdates.displayUpcomingRecipes(currentUser,recipeDatas)
+        menuMyUpcomingRecipes.classList.add('clicked')
+      }
+      else{
+        menuMyUpcomingRecipes.classList.remove('clicked')
+    domUpdates.displayAllRecipes(recipeDatas,currentUser)
+      }
+})
+let pantry;
 
 
 recipeCards.addEventListener('click', () => {
-if(event.target.classList.contains('star-icon')){
-  toogleFavorites(event)
+if(event.target.classList.contains('star-icon')) {
+  toggleFavorites(event)
+}
+if (event.target.classList.contains('plus-icon')) {
+  addToUpcomingRecipes(currentUser, recipeDatas);
 }
 })
-function toogleFavorites(event){
+
+function toggleFavorites(event){
   if(event.target.src === "https://image.flaticon.com/icons/svg/149/149222.svg"){
     event.target.src =   "https://image.flaticon.com/icons/svg/148/148841.svg"
     addToFavorites(event)
-    console.log(favorites)
   }else{   
     removeFromFavorites(event)
     event.target.src =  "https://image.flaticon.com/icons/svg/149/149222.svg" }
 }
 function addToFavorites(event){
```

Comment:
Both `addToFavorites()` and `removeFromFavorites()` should be refactored so that we're using the data model for `User` (stored as `currentUser`).  These methods exist on the `User` already.